scripts/scripts.js

```diff
@@ -354,6 +354,39 @@ function decorateH3(main) {
     });
   }
 }
+
+/**
+ * Checks for the same domain or not and if ends with pdf
+ */
+function isSameDomainOrPdf(url) {
+  // Create an anchor element to parse the URL
+  const ancElement = document.createElement('a');
```

Comment:
I would not create DOM elements for the purpose of checking the hostname. There is the JS class: `URL` for this purpose.

so on the original link element you can do `const url = new URL(link.href);`

and then you can do `url.hostname` or `url.pathname`.

Reply:
Avoided to create DOM elements by tweaking the logic in terms of absolute and relative urls.